provider/provider_test.go

```diff
@@ -458,3 +458,93 @@ func TestCheck(t *testing.T) {
 		})
 	}
 }
+
+func TestCanPreview(t *testing.T) {
```

Comment:
love this!

Reply:
It's a little more work up front to crank out some unit tests but worth it for the peace of mind!